.github/workflows/docker_image.yml

```diff
@@ -0,0 +1,52 @@
+# This workflow should be triggered when a release is done (workflow_call)
+# And also need to be able to run manually (workflow_dispatch). It uses
+# the current version from poetry as a tag for the docker container and 
+# pushes the container to ghcr.
+
+name: "Publish to GHCR"
+
+on: [workflow_call, workflow_dispatch]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: downcase REPO
+        run: |
+          echo "IMAGE_NAME=${IMAGE_NAME,,}" >>${GITHUB_ENV}
+      - name: checkout code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.VERSION_DECOIMPACT || github.token }}
+          fetch-depth: 0
+      - name: install python
```

Comment:
some of these steps are also in version_upgrade.yml. Can we remove the duplication?

Reply:
I tried. 
I could remove the duplication of the actually bump code. 
The thing is when using reusable workflows the main workflow can only use this in  a seperate job... but if you do some install in one job, it is not availbel in the next job. So I needed to do the checkout/install python/install poetry step a few more times for different jobs in order to make some of the code reusable.